polytracker/build.py

```diff
@@ -1,58 +1,384 @@
+
+    CXX_INCLUDE_PATH: Path = _cxx_dir_path() / "clean_build" / "include" / "c++" / "v1"
+    CXX_INCLUDE_PATH_ABI: Path = CXX_INCLUDE_PATH / "include" / "c++" / "v1"
+    CXX_LIB_PATH: Path = _cxx_dir_path() / "clean_build" / "lib"
+
+    LINK_LIBS: List[str] = [
+        str(CXX_LIB_PATH / "libc++.a"),
+        str(CXX_LIB_PATH / "libc++abi.a"),
+        "-lpthread",
+    ]
+
+    common_flags = ["-fPIC"]
+    linker_flags = ["-Wl,--start-group", *LINK_LIBS, "-Wl,--end-group"]
+    os.putenv("BLIGHT_ACTIONS", "InjectFlags:Record:FindInputs:FindOutputs:IgnoreFlags")
+    os.putenv(
+        "BLIGHT_ACTION_IGNOREFLAGS",
+        "FLAGS='-Wall -Wextra -Wno-unused-parameter -Werror'",
+    )
+    os.putenv(
+        "BLIGHT_ACTION_INJECTFLAGS",
+        f"CFLAGS='{' '.join(common_flags)}' "
+        f"CFLAGS_LINKER='{' '.join(linker_flags)}' "
+        f"CXXFLAGS='{' '.join(common_flags)} -stdlib=libc++ -I{CXX_INCLUDE_PATH!s} -I{CXX_INCLUDE_PATH_ABI!s}' "
+        f"CXXFLAGS_LINKER='{' '.join(linker_flags)} -L{CXX_LIB_PATH!s}'",
+    )
+    # Copy artifacts to $ARTIFACT_STORE_PATH_ENV and store their info in `blight_journal.jsonl`
+    if blight_journal.exists():
+        blight_journal.unlink()
+    os.putenv("BLIGHT_JOURNAL_PATH", str(blight_journal.absolute()))
+    os.putenv(
+        "BLIGHT_ACTION_FINDINPUTS",
+        f"store={ARTIFACT_STORE_PATH} append_hash=false",
+    )
+    os.putenv(
+        "BLIGHT_ACTION_FINDOUTPUTS",
+        f"store={ARTIFACT_STORE_PATH} append_hash=false",
+    )
+    os.putenv("BLIGHT_WRAPPED_CC", "gclang")
+    os.putenv("BLIGHT_WRAPPED_CXX", "gclang++")
+
+    subprocess.check_call(
+        ["blight-exec", "--guess-wrapped", "--swizzle-path", "--", *build_cmd]
+    )
+
+
+def _lower_bitcode(
+    input_bitcode: Path,
+    output_file: Path,
+    blight_cmd: Dict,
+) -> None:
+    POLY_LIB_PATH: Path = _ensure_path_exists(
+        _compiler_dir_path() / "lib" / "libPolytracker.a"
+    )
+    POLYCXX_LIBS: List[str] = [
+        str(_cxx_dir_path() / "poly_build" / "lib" / "libc++.a"),
+        str(_cxx_dir_path() / "poly_build" / "lib" / "libc++abi.a"),
+        str(POLY_LIB_PATH),
+        "-lm",
+        "-ltinfo",
+        "-lstdc++",
+    ]
+
+    DFSAN_LIB_PATH_ENV: str = _ensure_env_set("DFSAN_LIB_PATH")
+    DFSAN_LIB_PATH: Path = _ensure_path_exists(Path(DFSAN_LIB_PATH_ENV))
+
+    blight_record = blight_cmd["Record"]
+    blight_inputs = blight_cmd["FindInputs"]["inputs"]
+    blight_outputs = blight_cmd["FindOutputs"]["outputs"]
+    # Get the compiler
+    tool = blight_record["wrapped_tool"]
+    # Get source inputs to the original build command
+    inputs = list(map(lambda i: i["prenormalized_path"], blight_inputs))
+    # Get output of the original build command
+    outputs = list(map(lambda o: o["prenormalized_path"], blight_outputs))
+    # Get static libraries used for linking
+    libs = list(filter(lambda i: i["kind"] in ["static", "shared"], blight_inputs))
```

Comment:
should `"shared"` be included in the filter? The comment suggests otherwise.

Reply:
Yeah, I think this is a case of a wrong comment. I think dynamic libs are needed for some of the docker examples.